### DISCUSSION TITLE
How can I systematically learn OpenSSL step by step?

### DISCUSSION THREAD

**Discussion by NewYorkMen** [openssl/openssl#24142] [General Discussion]
Does anyone have relevant experience, such as where should I start, what documents should I read, etc., to comprehensively master what OpenSSL already offers?

**Comment by nhorman** [comment]
There is no canonical guide to everything openssl is capable of I'm afraid, but there are many resources available to you to both give you a general overview of the library, and references to accomplish specific tasks.

As guidance I would suggest the following resources:
1) Our [Youtube](https://www.youtube.com/@OpenSSL_) channel - Provides a video series we are actively expanding to give you a course style introduction to various openssl functional areas
2) Our [Manpages](https://openssl.org/docs/) - Provides a searchable API reference for the openssl libraries, along with various high level guides on using specific APIs
3) The [openssl wiki](https://wiki.openssl.org/index.php/Main_Page) - Provides  some more in depth discussion regarding various openssl APIs
4) Our [Demo code](https://github.com/openssl/openssl/tree/master/demos) - Provides larger examples of API usage, with commentary in the code
5) This [discussion page](https://github.com/openssl/openssl/discussions) - Provides a searchable archive of questions people have asked previously that are likely to cover the questions you might have in the future

**Comment by mattcaswell** [comment]
I would add that the [OpenSSL Guide pages](https://www.openssl.org/docs/man3.3/man7/ossl-guide-introduction.html) (part of the manpages @nhorman linked to above) are a good place to start. The guide provides overview and tutorial style introductory material to OpenSSL.

**Reply by NewYorkMen** [reply]
thank you very much!
